src/puj/push/instruction.clj

```diff
@@ -0,0 +1,228 @@
+(ns puj.push.instruction
+  (:require [clojure.spec.alpha :as s]
+            [cuerdas.core :as str]
+            [puj.push.pushstate :as pushstate]
+            [puj.util :as u]))
+
+(s/def ::name keyword?)
+
+(s/def ::code-blocks nat-int?)
+
+(s/def ::logic-fn fn?)
+
+(s/def ::stack-name keyword?)
+
+(s/def ::stack-list (s/coll-of ::stack-name))
+
+(s/def ::touched-stacks ::stack-list)
+
+(s/def ::input-stacks ::stack-list)
+
+(s/def ::output-stacks ::stack-list)
+
+(s/def ::other-stacks ::stack-list)
```

Comment:
Have to ask for an example here. Are there instructions with "side-effects", which I assume this is for, or is it "just in case"?

Reply:
This is common in the instructions of PyshGP and Clojush. For example, `string-stack-depth` requires a string stack to exist, but it doesn't consume or push items to it.